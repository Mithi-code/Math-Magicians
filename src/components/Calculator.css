.calculator {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
}

.result {
  background-color: rgb(189, 185, 216);
  width: 400px;
  height: 50px;
  display: flex;
  padding: 0 5px;
  align-items: center;
  justify-content: end;
  color: #fff;
  font-size: 20px;
}

table {
  width: 400px;
  border: 1px solid rgb(160, 157, 157);
}

tr {
  border: 1px solid rgb(160, 157, 157);
  height: 50px;
  background-color: #ebe8e8;
}

button {
  outline: none;
  border: none;
  background-color: unset;
  width: 100%;
  font-size: 20px;
}

td {
  height: 40px;
  width: 50px;
}

td:last-child {
  background-color: #ffa500;
}
