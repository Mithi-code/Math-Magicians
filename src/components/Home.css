* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.text {
  padding: 45px;
  width: 600px;
  font-size: 18px;
  color: azure;
}

h2 {
  margin-bottom: 30px;
}

p {
  font-size: 20px;
}

.btn {
  margin-top: 20px;
  width: 150px;
  padding: 5px;
  font-size: 22px;
  background-color: aqua;
}
