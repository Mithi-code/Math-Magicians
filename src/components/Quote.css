* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.quote-text {
  padding: 15px;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 600px;
  margin: auto;
  font-size: 25px;
  border: 2px solid aqua;
  border-radius: 20px;
  color: aliceblue;
}
