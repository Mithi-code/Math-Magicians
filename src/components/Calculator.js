import React, { Component } from 'react';
import './Calculator.css';

// eslint-disable-next-line react/prefer-stateless-function
class Calculator extends Component {
  render() {
    return (
      <div className='calculator'>
        <div className='result'>0</div>
        <table>
          <tr>
            <td>AC</td>
            <td>+/-</td>
            <td>%</td>
            <td>+</td>
          </tr>
          <tr>
            <td>7</td>
            <td>8</td>
            <td>9</td>
            <td>Ã—</td>
          </tr>
          <tr>
            <td>4</td>
            <td>5</td>
            <td>6</td>
            <td>-</td>
          </tr>
          <tr>
            <td>1</td>
            <td>2</td>
            <td>3</td>
            <td>+</td>
          </tr>
          <tr>
            <td colSpan={2}>0</td>
            <td>.</td>
            <td>=</td>
          </tr>
        </table>
      </div>
    );
  }
}

export default Calculator;
