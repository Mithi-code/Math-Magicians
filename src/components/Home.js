import React from 'react';
import './Home.css';

function Home() {
    return (
        <div>
            <section className="text">
                <h2>The Perfect way to master Mathematicsis to practice Mathematics</h2>
                <p>Start learning mathematics using free tool calculator.</p>
                <a href="/Calculator">
                    <button className="btn" type="button">Click To Start!</button>
                </a>
            </section>
        </div>
    )
}

export default Home
