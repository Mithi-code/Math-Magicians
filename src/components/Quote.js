import React from 'react';
import './Quote.css';

function Home() {
    return (
        <div>
           
           <div className="container">
               <h3 className="quote-text">
               Mathematics as an expression of the human mind reflects the active will, the contemplative reason, and the desire for aesthetic perfection. Its basic elements are logic and intuition, analysis and construction, generality and individuality.
               â€” Richard Courant.
               </h3>
           </div>
        </div>
    )
}

export default Home
