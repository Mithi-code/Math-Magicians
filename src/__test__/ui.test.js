import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Calculator from '../components/Calculator';
import Quote from '../components/Quote';
import Home from '../components/Home';

describe('Articles Tests', () => {
  it('Render works correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Home />

      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Render works fine', () => {
    const tree = renderer.create(<Calculator />);
    expect(tree).toMatchSnapshot();
  });

  it('Render works good', () => {
    const tree = renderer.create(<Quote />);
    expect(tree).toMatchSnapshot();
  });
});