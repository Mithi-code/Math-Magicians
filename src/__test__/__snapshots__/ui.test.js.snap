// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles Tests Render works correctly 1`] = `
<div>
  <section
    className="text"
  >
    <h2>
      The Perfect way to master Mathematicsis to practice Mathematics
    </h2>
    <p>
      Start learning mathematics using free tool calculator.
    </p>
    <a
      href="/Calculator"
    >
      <button
        className="btn"
        type="button"
      >
        Click To Start!
      </button>
    </a>
  </section>
</div>
`;

exports[`Articles Tests Render works correctly 2`] = `
<div
  className="calculator"
>
  <div
    className="result"
  >
    0
  </div>
  <table>
    <tr>
      <td>
        <button
          name="AC"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
      </td>
      <td>
        <button
          name="+/-"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
      </td>
      <td>
        <button
          name="%"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
      </td>
      <td>
        <button
          name="÷"
          onClick={[Function]}
          type="button"
        >
          ÷
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          name="7"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
      </td>
      <td>
        <button
          name="8"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
      </td>
      <td>
        <button
          name="9"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
      </td>
      <td>
        <button
          className="operator"
          name="x"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          name="4"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
      </td>
      <td>
        <button
          name="5"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
      </td>
      <td>
        <button
          name="6"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
      </td>
      <td>
        <button
          name="-"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </td>
    </tr>
    <tr>
      <td>
        <button
          name="1"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
      </td>
      <td>
        <button
          name="2"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
      </td>
      <td>
        <button
          name="3"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
      </td>
      <td>
        <button
          name="+"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </td>
    </tr>
    <tr>
      <td
        colSpan={2}
      >
        <button
          name="0"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
      </td>
      <td>
        <button
          name="."
          onClick={[Function]}
          type="button"
        >
          .
        </button>
      </td>
      <td>
        <button
          name="="
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </td>
    </tr>
  </table>
</div>
`;

exports[`Articles Tests Render works correctly 3`] = `
<div>
  <div
    className="container"
  >
    <h3
      className="quote-text"
    >
      Mathematics as an expression of the human mind reflects the active will, the contemplative reason, and the desire for aesthetic perfection. Its basic elements are logic and intuition, analysis and construction, generality and individuality. — Richard Courant.
    </h3>
  </div>
</div>
`;
